zammad/templates/pvc.yaml

```diff
@@ -0,0 +1,21 @@
+{{ if and .Values.persistence.enabled (not .Values.persistence.existingClaim) }}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ template "zammad.fullname" . }}-var
+spec:
+  accessModes:
+  {{- range .Values.persistence.accessModes }}
+    - {{ . | quote }}
+  {{- end }}
+  resources:
+    requests:
+      storage: {{ .Values.persistence.size | quote }}
+  {{- if .Values.persistence.storageClass }}
+  {{- if (eq "-" .Values.persistence.storageClass) }}
+  storageClassName: ""
+  {{- else }}
+  storageClassName: "{{ .Values.persistence.storageClass }}"
+  {{- end }}
+  {{- end }}
+{{ end }}
```

Comment:
Is this the right way to handle the volume, if no external claim is provided?
It will now use `ReadWriteMany` by default, and the underlying `PersistentVolume` has a `ReclaimPolicy` of `Delete`. This means the data will be lost when the application is uninstalled.

An alternative might be to always require an external claim to be provided when provisioning the application. This would force the admins to make their choice first.

Reply:
We will change this so that an external claim is required.